Modules/_tracemalloc.c

```diff
@@ -78,15 +78,18 @@ __attribute__((packed))
 
 typedef struct {
     Py_uhash_t hash;
-    int nframe;
+    /* Number of frames stored */
+    uint16_t nframe;
+    /* Total number of frames the traceback had */
+    uint16_t total_nframe;
     frame_t frames[1];
 } traceback_t;
 
 #define TRACEBACK_SIZE(NFRAME) \
         (sizeof(traceback_t) + sizeof(frame_t) * (NFRAME - 1))
 
 #define MAX_NFRAME \
-        ((INT_MAX - (int)sizeof(traceback_t)) / (int)sizeof(frame_t) + 1)
+        ((UINT16_MAX - (int)sizeof(traceback_t)) / (int)sizeof(frame_t) + 1)
```

Comment:
@vstinner I'm not really sure about this change. I actually can't understand how `MAX_NFRAME` is calculated here. Can you give me a hint?

Reply:
It looks like I wrote to ensure that TRACEBACK_SIZE(nframe) will not overflow, but I am not sure that using INT_MAX makes sense.

TRACEBACK_SIZE() maximum should be smaller than or equal to SIZE_MAX, not INT_MAX.

Maybe Py_MIN(INT_MAX, ((SIZE_MAX - sizeof(traceback_t)) / sizeof(frame_t) + 1) would be more correct?

Reply:
> TRACEBACK_SIZE() maximum should be smaller than or equal to SIZE_MAX, not INT_MAX.

Agreed. `TRACEBACK_SIZE()` is actually a `size_t`.

> Py_MIN(INT_MAX, ((SIZE_MAX - sizeof(traceback_t)) / sizeof(frame_t) + 1) would be more correct?

Yes. I think you either want the maximum size you can allocate or the maximum number of frame you can store in a `uint16_t`.

I'll change that.